swift/src/main/kotlin/com.r3.corda.finance.swift/services/SWIFTClient.kt

```diff
@@ -48,6 +59,56 @@ class SWIFTClient(
                     creditorBicfi : String,
                     remittanceInformation : String
     ) : SWIFTPaymentResponse {
+
+        val paymentResponse = submitPaymentInstruction(e2eId,
+                executionDate,
+                amount,
+                debtorName,
+                debtorLei,
+                debtorIban,
+                debtorBicfi,
+                creditorName,
+                creditorLei,
+                creditorIban,
+                creditorBicfi,
+                remittanceInformation)
+
+        val unsignedPayload = getUnsignedPayload(paymentResponse.uetr)
```

Comment:
Looks like this is where all the magic happens.. We first get a payload which constitutes the payment initiation... Then we sign it and re-submit it and just return the initial response ?

Reply:
Yeps, exactly right. The rest of the process is effectively the same.